src/sentry/toolbar/views/iframe_view.py

```diff
@@ -15,77 +16,63 @@ def _get_referrer(request) -> str | None:
     return request.META.get("HTTP_REFERER")
 
 
-@region_silo_view
-class IframeView(ProjectView):
-    def respond(self, template: str, context: dict[str, Any], status: int = 200):
-        self.default_context = {}  # self.get_context_data(request, *args, **kwargs)
+def _add_frame_headers(request: HttpRequest, response: HttpResponseBase):
+    referrer = _get_referrer(request) or ""
 
-        referrer = context.get("referrer") or "'none'"
-        response = super().respond(template, context=context, status=status)
+    # This is an alternative to @csp_replace - we need to use this pattern to access the referrer.
+    response._csp_replace = {"frame-ancestors": [referrer.strip("/") or "'none'"]}  # type: ignore[attr-defined]
+    response["X-Frame-Options"] = "DENY" if referrer == "" else "ALLOWALL"
+    return response
 
-        # This is an alternative to @csp_replace - we need to use this pattern to access the referrer.
-        response._csp_replace = {"frame-ancestors": [referrer.strip("/")]}  # type: ignore[attr-defined]
-        response["X-Frame-Options"] = "DENY" if referrer == "'none'" else "ALLOWALL"
 
-        return response
+@region_silo_view
+class IframeView(ProjectView):
+    default_context = {}
 
-    def handle_disabled_member(self, request: HttpRequest, *args, **kwargs):
-        return self._handle_logged_out(request, *args, **kwargs)
+    def dispatch(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponseBase:
+        self.organization_slug = kwargs.get("organization_slug", "")
+        self.project_id_or_slug = kwargs.get("project_id_or_slug", "")
+        return super().dispatch(request, *args, **kwargs)
 
-    def handle_not_2fa_compliant(self, request: HttpRequest, *args, **kwargs):
-        return self._handle_logged_out(request, *args, **kwargs)
+    def handle_disabled_member(self, organization: Organization) -> HttpResponse:
```

Comment:
```suggestion
    def handle_disabled_member(self, *args, **kwargs) -> HttpResponse:
```
Makes it clear we don't care about the args, just matching the superclass

Reply:
the superclass defines this with only organization as an arg:

https://github.com/getsentry/sentry/blob/1316bc7d0f8e805c0c40b95b920f08e31e9ff17f/src/sentry/web/frontend/base.py#L495-L497

Reply:
Yeah, but since it's unused anyway, I was saying the args, kwargs will match any pattern. It'sa nit. Can leave as-is